tiledb/sm/c_api/tiledb.cc

```diff
@@ -4956,7 +4957,25 @@ int32_t tiledb_fragment_info_get_fragment_name(
       sanity_check(ctx, fragment_info) == TILEDB_ERR)
     return TILEDB_ERR;
 
-  throw_if_not_ok(fragment_info->fragment_info_->get_fragment_name(fid, name));
+  *name = (new std::string(fragment_info->fragment_info_->fragment_name(fid)))
+              ->c_str();
+
+  return TILEDB_OK;
+}
+
+int32_t tiledb_fragment_info_get_fragment_name_v2(
+    tiledb_ctx_t* ctx,
+    tiledb_fragment_info_t* fragment_info,
+    uint32_t fid,
+    tiledb_string_t** name) {
+  if (sanity_check(ctx, fragment_info) == TILEDB_ERR)
+    return TILEDB_ERR;
+
+  if (name == nullptr)
+    throw std::invalid_argument("Name cannot be null.");
```

Comment:
Here and above.
Style: Always use `{}` with `if`, even for one-line conditionals.